src/Composer/Installers/CakePHPInstaller.php

```diff
@@ -58,25 +58,15 @@ protected function matchesCakeVersion($matcher, $version)
         }
 
         $repositoryManager = $this->composer->getRepositoryManager();
-        if ($repositoryManager) {
-            $repos = $repositoryManager->getLocalRepository();
-            if (!$repos) {
-                return false;
-            }
-            $cake3 = new $multiClass(array(
-                new $constraintClass($matcher, $version),
-                new $constraintClass('!=', '9999999-dev'),
-            ));
-            $pool = new Pool('dev');
-            $pool->addRepository($repos);
-            $packages = $pool->whatProvides('cakephp/cakephp');
-            foreach ($packages as $package) {
-                $installed = new $constraintClass('=', $package->getVersion());
-                if ($cake3->matches($installed)) {
-                    return true;
-                }
-            }
+        if (! $repositoryManager) {
```

Comment:
So all this looks to be asking if `cakephp/cakephp` is >= 3, but in a complicated way that touched a lot of Composer's APIs and breaks against Composer 2. This way to answer the same question is compatible with `^1.0|^2.0`